src/vs/platform/terminal/node/terminalProfiles.ts

```diff
@@ -160,10 +160,12 @@ async function transformToTerminalProfiles(
 			} else if (source.icon) {
 				icon = source.icon;
 			}
-		} else {
+		} else if ('path' in profile) {
```

Comment:
I don't think we want the possibility of `IExtensionTerminalProfile` objects reaching here?

Reply:
they won't - it's just that `ITerminalProfileObject` (the return type) includes the `IExtensionTerminalProfile`

Reply:
We should avoid including `IExtensionTerminalProfile` in the union type then. I've never been that happy with that name, how about this to help clarify?

```ts
export type IUnresolvedTerminalProfile = ITerminalExecutable | ITerminalProfileSource | null;
```

So `IExtensionTerminalProfile` would be considered a resolved terminal profile since it's already filled and no longer needs validation (unlike path/source/etc. from `IUnresolvedTerminalProfile`).